code.user.js

```diff
@@ -455,15 +458,18 @@
                 amount: 1,
                 price: price
             },
-            success: function(data) {
-                if (data.success === false && isRetryMessage(data.message)) {
+            success: function(data, status, xhr) {
+                marketRateLimitReached = xhr.status === 429;
+
+                if (data && data.success === false && isRetryMessage(data.message)) {
                     callback(ERROR_FAILED, data);
                 } else {
                     callback(ERROR_SUCCESS, data);
                 }
             },
-            error: function(data) {
-                return callback(ERROR_FAILED, data);
+            error: function(xhr) {
+                marketRateLimitReached = xhr.status === 429;
+                return callback(ERROR_FAILED, xhr);
```

Comment:
Passing xhr seems wrong (you didn't change it I know), probably meant to be passing the response text or something? Could you verify what the callback expects on error please.

Reply:
> Passing xhr seems wrong (you didn't change it I know), probably meant to be passing the response text or something? Could you verify what the callback expects on error please.

yes, ig it's not correct
i changed this behaviour on last commits